commercetools/resource_project.go

```diff
@@ -280,31 +288,41 @@ func projectUpdate(d *schema.ResourceData, client *commercetools.Client, version
 
 	if d.HasChange("carts") {
 		carts := d.Get("carts").(map[string]interface{})
+		var fallbackEnabled bool = false
 		if carts["country_tax_rate_fallback_enabled"] != nil {
 			// boolean value is somehow interface{} | string so....
-			var fallbackEnabled bool
 			switch carts["country_tax_rate_fallback_enabled"] {
 			case "true":
 				fallbackEnabled = true
 			case "false":
-				fallbackEnabled = false
 			default:
 				return fmt.Errorf("invalid value for carts[\"country_tax_rate_fallback_enabled\"]: %s", carts["country_tax_rate_fallback_enabled"])
 			}
-			input.Actions = append(
-				input.Actions,
-				&commercetools.ProjectChangeCountryTaxRateFallbackEnabledAction{
-					CountryTaxRateFallbackEnabled: fallbackEnabled,
-				})
-		} else {
-			// To commercetools this field is not optional, so when deleting we revert to the default: false:
-			input.Actions = append(
-				input.Actions,
-				&commercetools.ProjectChangeCountryTaxRateFallbackEnabledAction{
-					CountryTaxRateFallbackEnabled: false,
-				})
 		}
 
+		input.Actions = append(
+			input.Actions,
+			&commercetools.ProjectChangeCountryTaxRateFallbackEnabledAction{
+				CountryTaxRateFallbackEnabled: fallbackEnabled,
+			})
+
+		var deleteDaysAfterLastModification int
+		if carts["delete_days_after_last_modification"] != nil {
+			// boolean value is somehow interface{} | string so....
```

Comment:
This comment is not relevant here and can be removed

Reply:
I'll change it to say int value, as it's still referring to the same issue where objects don't reflect the correct types and everything returns as a string. I looked into fixing this as if you use schema.TypeList instead of TypeMap the values are reflected correctly, but TypeMap seems to destroy all type mappings and returns everything as a string instead. My go knowledge isn't good enough to understand and fix this sadly

Reply:
Interesting. This might be fixable by upgrading to the v2 SDK, I think they changed the structure of TypeMap a little in that, though I know it's an ongoing discussion how flexible TypeMap should be. That's on my to-do list somewhere down the line.

Anyway thanks for the quick fix. Looks good to me. I've mergen and Ill make a new release later today after potentially merging another open PR and a few chores.

Reply:
Hopefully, it's annoying to have all these type conversions everywhere. Thanks for the quick review and merge! :)